<launch>

  <arg name="map_file" default="$(find light_scan_sim)/data/empty.yaml"/>

  <!-- Run the map server -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />

  <!-- Publish SegmentList -->
  <node pkg="rostopic" type="rostopic" name="publish_circle_list" args="pub --latch /map_circles light_scan_sim/CircleList -f $(find light_scan_sim)/data/circle.yaml" output="screen" />
  <!-- Publish Materials -->
  <node pkg="rostopic" type="rostopic" name="publish_material_list" args="pub --latch /map_materials light_scan_sim/MaterialList -f $(find light_scan_sim)/data/materials_circle.yaml" output="screen" />

  <!-- publish a tf at pose estimate position -->
  <node name="pose_estimate_publisher" pkg="light_scan_sim" type="pose_estimate_tf_publisher.py" output="screen" />

  <!-- visualize the segment list -->
  <node name="visualize_circle_list" pkg="light_scan_sim" type="visualize_circle_list.py" output="screen">
    <param name="input_topic" value="/map_circles" />
    <param name="materials_topic" value="/map_materials" />
    <param name="output_topic" value="/map_circle_vis" />
  </node>

  <!-- The scan simulator -->
  <node name="light_scan_sim" pkg="light_scan_sim" type="light_scan_sim_node" output="screen">
    <rosparam file="$(find light_scan_sim)/config/config.yaml" command="load" />
  </node>

  <!-- RViz -->
  <node name="visualisation" pkg="rviz" type="rviz" args="-d $(find light_scan_sim)/data/test.rviz" />

</launch>
